subprojects/gst-plugins-bad/sys/msdk/gstmsdkenc.c

```diff
@@ -146,13 +146,24 @@ gst_msdkenc_set_context (GstElement * element, GstContext * context)
     gst_object_replace ((GstObject **) & thiz->context,
         (GstObject *) msdk_context);
     gst_object_unref (msdk_context);
-  } else if (gst_msdk_context_from_external_display (context,
+  } else
+#ifndef _WIN32
+    if (gst_msdk_context_from_external_display (context,
           thiz->hardware, 0 /* GST_MSDK_JOB_ENCODER will be set later */ ,
           &msdk_context)) {
     gst_object_replace ((GstObject **) & thiz->context,
         (GstObject *) msdk_context);
     gst_object_unref (msdk_context);
   }
+#else
+    if (gst_msdk_context_from_external_device (context,
+          thiz->hardware, 0 /* GST_MSDK_JOB_ENCODER will be set later */ ,
+          &msdk_context)) {
+    gst_object_replace ((GstObject **) & thiz->context,
+        (GstObject *) msdk_context);
+    gst_object_unref (msdk_context);
+  }
+#endif
```

Comment:
`gst_msdk_context_from_external_display` and `gst_msdk_context_from_external_device` have the same arguments. the {} blocks are same too, may we replace `gst_msdk_context_from_external_device` and `gst_msdk_context_from_external_device` with the same function name ?  We may have different implementation on Linux and Windows.

Reply:
I added a priv->device for Win. The gst_msdk_context_get_device and gst_msdk_context_from_external_device are for the use of priv->device. Do you mean I delete priv->device and keep priv->display for Win? I thought display meant va display.